src/utils.c

```diff
@@ -424,3 +424,102 @@ SEXP startsWithAny(const SEXP x, const SEXP y, SEXP start) {
   return ScalarLogical(false);
 }
 
+SEXP frev(SEXP x, SEXP copyArg) {
+  if (INHERITS(x, char_dataframe) || INHERITS(x, char_datatable))
+    error(_("'x' should not be data.frame or data.table."));
+  if (!isNull(getAttrib(x, R_DimSymbol)))
+    error(_("'x' should not be matrix or array"));
+  if (!IS_TRUE_OR_FALSE(copyArg))
+    error(_("%s must be TRUE or FALSE."), "copy");
+  bool copy = LOGICAL(copyArg)[0];
+  int n = LENGTH(x);
+  int nprotect = 0;
+  if (copy) {
+    x = PROTECT(duplicate(x));
+    nprotect++;
+    copy = false;
+  }
+  if (n==0) {
+    UNPROTECT(nprotect);
+    return x;
+  }
+  switch (TYPEOF(x)) {
+    case LGLSXP: case INTSXP: {
+      int *restrict xd = INTEGER(x);
```

Comment:
I believe `INTEGER` is not permitted on type `LGLSXP`, even if the underlying type is `int`

Reply:
We also have this in `subset.c` and `gsumm.c` so might want to remove it there too!

Reply:
Then, I don't think it is actually necessary. It should have been bite us before.